data_structures/src/serializers.rs

```diff
@@ -0,0 +1,572 @@
+struct VerackWitnetArgs {
+    magic: u16,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct VersionFlatbufferArgs<'a> {
+    capabilities: u64,
+    genesis: u64,
+    last_epoch: u32,
+    magic: u16,
+    nonce: u64,
+    receiver_address: &'a Address,
+    sender_address: &'a Address,
+    timestamp: u64,
+    user_agent: &'a str,
+    version: u32,
+}
+
+#[derive(Debug, Clone)]
+struct VersionWitnetArgs {
+    capabilities: u64,
+    genesis: u64,
+    last_epoch: u32,
+    magic: u16,
+    nonce: u64,
+    receiver_address: Address,
+    sender_address: Address,
+    timestamp: u64,
+    user_agent: String,
+    version: u32,
+}
+
+pub trait FromBytes<T> {
+    fn from_bytes(bytes: Vec<u8>) -> Option<T>;
+}
+
+pub trait ToBytes {
+    fn to_bytes(&self) -> Vec<u8>;
+}
+
+// #[cfg(feature = "flatbuffers")]
+impl FromBytes<Message> for Message {
+    fn from_bytes(bytes: Vec<u8>) -> Option<Self> {
+        // Get Flatbuffers Message
+        let message = get_root_as_message(&bytes);
+        let magic = message.magic();
+
+        // Create witnet's message to decode a flatbuffer message
+        match message.command_type() {
+            FlatBuffersCommand::Ping => message.command_as_ping().map(|ping| {
+                create_ping_message(&PingWitnetArgs {
+                    nonce: ping.nonce(),
+                    magic,
+                })
+            }),
+            FlatBuffersCommand::Pong => message.command_as_pong().map(|pong| {
+                create_pong_message(&PongWitnetArgs {
+                    nonce: pong.nonce(),
+                    magic,
+                })
+            }),
+            FlatBuffersCommand::GetPeers => {
+                Some(create_get_peers_message(GetPeersWitnetArgs { magic }))
+            }
+            FlatBuffersCommand::Peers => message
+                .command_as_peers()
+                .and_then(|peers| create_peers_message(PeersWitnetArgs { magic, peers })),
+            FlatBuffersCommand::Verack => Some(create_verack_message(VerackWitnetArgs { magic })),
+            FlatBuffersCommand::Version => message.command_as_version().and_then(|command| {
+                // Get ftb addresses and create witnet addresses
+                let sender_address = command.sender_address().and_then(create_address);
+                let receiver_address = command.receiver_address().and_then(create_address);
+                // Check if sender address and receiver address exist
+                if sender_address.and(receiver_address).is_some() {
+                    Some(create_version_message(VersionWitnetArgs {
+                        version: command.version(),
+                        timestamp: command.timestamp(),
+                        capabilities: command.capabilities(),
+                        sender_address: sender_address.unwrap(),
+                        receiver_address: receiver_address.unwrap(),
+                        // user_agent field should be required as specified in ftb schema
```

Comment:
Please, mark this as TODO so that we can easily locate this in the future

Reply:
personally, I'd like the workflow of `FIXME/TODO` in `rust-lang/rust`:
 1. use ***temporary*** `TODO` marker in commits;
2. the contributor and reviewer work together to clean up the `TODO`s as more as possible;
3. and turn unfixed `TODO`s into ***permanent*** `FIXME`;
4. ensure no TODO in the PR before final merging(`rust-lang/rust` ensures this by rejecting `TODO` in CI).

And the format of `FIXME`:
```
// FIXME(issue-number ): must have comments 

//snippet from rustc codebase

// FIXME: this should be `cannot borrow` (issue #18330)
let node_id = item.as_local_node_id();
```

the comments and issue-number is ***must-have***, and the unfixed case should be exposed with test case if possible.